franklin/builder/models.py

```diff
@@ -92,40 +89,62 @@ class Build(models.Model):
 
     :param site: Reference to the project for this build instance
     :param git_hash: If a branch build, the git hash of the deployed code
-    :param branch: If a branch build, the name of the branch
-    :param tag: If a tag build, the name of the tag
     :param created: Date this code was built
     :param path: The path of the site on the static server
     """
 
     site = models.ForeignKey(Site, related_name='builds')
-    git_hash = models.CharField(max_length=40, blank=True, null=True)
-    branch = models.CharField(max_length=100, blank=True, null=True)
-    tag = models.CharField(max_length=100, blank=True, null=True)
     created = models.DateTimeField(editable=False)
     path = models.CharField(max_length=100)
     
+
+class TagBuild(Build):
+    """ Flavor of build that was created from a tag
+    
+    :param tag: If a tag build, the name of the tag
+    """
+    tag = models.CharField(max_length=100, blank=True, null=True)
+
     def save(self, *args, **kwargs):
         base_path = os.environ['BASE_PROJECT_PATH']
-        if self.tag:
-            clean_tag = re.sub('[^a-zA-Z0-9]', '', self.tag)
-            self.path = "{0}/{1}/{2}/{3}".format(
-                base_path, self.site.owner.name, self.site.name, clean_tag)
-        else:
-            self.path = "{0}/{1}/{2}/{3}".format(
-                base_path, self.site.owner.name, self.site.name, self.git_hash)
+        clean_tag = re.sub('[^a-zA-Z0-9]', '', self.tag)
+        self.path = "{0}/{1}/{2}/{3}".format(
+            base_path, self.site.owner.name, self.site.name, clean_tag)
         if not self.id:
             self.created = timezone.now()
-        super(Build, self).save(*args, **kwargs)
+        super(TagBuild, self).save(*args, **kwargs)
+
+    def __str__(self):
+        return '%s %s' % (self.site.name, self.tag)
     
+    class Meta(object):
+        verbose_name = _('TagBuild')
+        verbose_name_plural = _('TagBuilds')
+
+
+class BranchBuild(Build):
+    """ Flavor of build that was created from a branch
+    
+    :param branch: If a branch build, the name of the branch
+    :param git_hash: If a branch build, the git hash of the deployed code
+    """
+    git_hash = models.CharField(max_length=40, blank=True, null=True)
+    branch = models.CharField(max_length=100, blank=True, null=True)
+
```

Comment:
now that these are new models, we can enforce these values to be required as well as unique together